### DISCUSSION TITLE
[Question] Env var precedence

### DISCUSSION THREAD

**Discussion by kobenguyent** [flawiddsouza/Restfox#268] [Q&A] [answered]
So I have a postman collection which has env vars set at collection, for instance `email` is `123@hello.com`, it is called `Variables` when opening the Collection panel
![Screenshot 2024-10-01 at 13 53 47](https://github.com/user-attachments/assets/642e869e-62b3-4d3d-9bca-e9821ee427a1)

Now when I choose an Environment from the Environment list, this Environment is also set the `email` is `345@hello.com`.

When calling a request which invoking this `email`
- on Postman: it is `345@hello.com`
- on Restfox: it is `123@hello.com`

And the question is, what shall be the higher precedence

**Comment by flawiddsouza** [comment, marked as answer]
Restfox has a defined order:
Workspace (`email: test@test.com`) -> Folder (`email: test2@test.com`) -> Sub Folder (`email: test3@test.com`) -> Resolved Request Env (`email: test3@test.com`)

Restfox resolves environment variables from top to bottom. So you can use a workspace env var inside a folder env var.

Workspace env:
```
test: 1
```

Folder env:
```
my_var: {{ test }}
```

Resolved Request Env:
```
my_var: 1
```

Postman's behavior seems incorrect, as immediate parent should take precedence.

I'll explain in code
```js
let a = 10 // workspace env
{ // folder env
    a = 20
    { // sub folder env
        a = 30
        {
            console.log(a) // a should be 30 and not 10
        }
    }
}
```
